plugins/FlashSales/src/EasyAbp.EShop.Plugins.FlashSales.Application/EasyAbp/EShop/Plugins/FlashSales/FlashSalePlans/RollBackInventoryCreateFlashSaleOrderCompleteEventHandler.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Threading.Tasks;
+using EasyAbp.Eshop.Products.Products;
+using EasyAbp.EShop.Plugins.FlashSales.FlashSaleResults;
+using EasyAbp.EShop.Products.Products;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Logging;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.EventBus.Distributed;
+using Volo.Abp.Uow;
+
+namespace EasyAbp.EShop.Plugins.FlashSales.FlashSalePlans;
+
+public class RollBackInventoryCreateFlashSaleOrderCompleteEventHandler : IDistributedEventHandler<CreateFlashSaleOrderCompleteEto>, ITransientDependency
+{
+    protected IFlashSaleInventoryManager FlashSaleInventoryManager { get; }
+    protected IDistributedCache DistributedCache { get; }
+    protected IFlashSalePlanRepository FlashSalePlanRepository { get; }
+    protected IProductAppService ProductAppService { get; }
+    protected ILogger<RollBackInventoryCreateFlashSaleOrderCompleteEventHandler> Logger { get; }
+
+    public RollBackInventoryCreateFlashSaleOrderCompleteEventHandler(
+        IFlashSaleInventoryManager flashSaleInventoryManager,
+        IDistributedCache distributedCache,
+        IFlashSalePlanRepository flashSalePlanRepository,
+        IProductAppService productAppService)
+    {
+        FlashSaleInventoryManager = flashSaleInventoryManager;
+        DistributedCache = distributedCache;
+        FlashSalePlanRepository = flashSalePlanRepository;
+        ProductAppService = productAppService;
+    }
+
+    [UnitOfWork(true)]
```

Comment:
Why create a UOW here?

Reply:
I didn't notice that 'DefaultProductInventoryProvider' has opened UOW. I will fix it.